### PR TITLE
Modification on the PAG community pipeline

### PR DESCRIPTION
**Pull request by HyoungwonCho**
# What does this PR do?

## Perturbed-Attention Guidance 

This implementation is based on [Diffusers](https://huggingface.co/docs/diffusers/index). StableDiffusionPAGPipeline is a modification of StableDiffusionPipeline to support Perturbed-Attention Guidance (PAG). For more information about PAG, please refer to [Project](https://ku-cvlab.github.io/Perturbed-Attention-Guidance/) / [arXiv](https://arxiv.org/abs/2403.17377) / [GitHub](https://github.com/KU-CVLAB/Perturbed-Attention-Guidance)

Hi! We have updated some of the implementations of the Perturbed-Attention Guidelines, which we previously contributed to community pipelines.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: sample.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import os
```
Comment (bghira) [inline sample.py:1]: did you mean to commit this file too?

Comment (HyoungwonCho) [reply sample.py:1]: Yes, this is the request I sent with some modifications to the code I previously committed. I would appreciate it if you could reflect it!

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by yiyixuxu** [comment]
can we run `make quality`?
will merge once the tests pass

**Comment by yiyixuxu** [comment]
cc @asomoza here

**Comment by asomoza** [comment]
Thanks for your work, this looks great, I didn't have the chance to be for the first PR so I'm just trying this one out.

I really need to sit down and test the effects of PAG because I find them really interesting, for example some of the blocks I often use for LoRAs:

|normal|scale 5.0, "m0"|scale=5.0, "d4", "d5", "m0"|scale=7.0, "d4", "d5", "m0"|
|---|---|---|---|
|||||

Still I think it would be better to use SDXL to showcase it, this one is supposed to be a "steam powered mechanic spider robot" but at least PAG makes it more of a "spider".

**Comment by HyoungwonCho** [comment]
@yiyixuxu Thank you, I removed the white space in the code for code quality check.

**Comment by HyoungwonCho** [comment]
@asomoza Thank you for your interest in our work!

Thanks to [multimodalart](https://github.com/multimodalart), users can try PAG on SDXL on Hugging Face Spaces.

For more information, please visit our [github page](https://github.com/KU-CVLAB/Perturbed-Attention-Guidance)

**Comment by HyoungwonCho** [comment]
I am struggling with build error. If this is resolved, I will open a new pull request.

**Comment by bghira** [comment]
@HyoungwonCho if there is anything i can help with, please ping me here or on discord

**Pull request closed without merging**